### DISCUSSION TITLE
how to include the detected maps/masks with the final image

### DISCUSSION THREAD

**Discussion by dave8x** [lllyasviel/ControlNet#563] [Q&A]
hi guys, i tried searching for similar question but couldnt find it.

I wish to know how to include the image used for controlnet in the final result. What I mean is, once I used a google colab stable diffusion which had this feature. For example you load an image in control net (for example as reference) then create a grid (xyz plot) from the txt2img or img2img, and the result grid will have 2 pictures for each entry. first picture is the result, and next to it is the controlnet map used (in this case the reference imaged). they'd be fused together as a single image for reference purpose. This also work for other preprocessors like lineart, scribble etc, in that case you will see the black with white outline mask. 

really wish to have it enabled as it helps me keep track what was used. I'm using A1111

thanks

**Comment by geroldmeisinger** [comment]
cross-post of https://github.com/Mikubill/sd-webui-controlnet/discussions/2189
